substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs

```diff
@@ -0,0 +1,263 @@
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<(), &'static str> {
+            assert!(PalletVersion::<T>::get() == types::StorageVersion::V9Struct);
+
+            info!("👥  TFGrid pallet to v10 passes PRE migrate checks ✅",);
+            Ok(())
+        }
+
+        fn on_runtime_upgrade() -> Weight {
+            migrate::<T>()
+        }
+
+        #[cfg(feature = "try-runtime")]
+        fn post_upgrade() -> Result<(), &'static str> {
+            assert!(PalletVersion::<T>::get() == types::StorageVersion::V10Struct);
+
+            info!(
+                "👥  TFGrid pallet migration to {:?} passes POST migrate checks ✅",
+                Pallet::<T>::pallet_version()
+            );
+
+            Ok(())
+        }
+    }
+}
+
+fn migrate<T: Config>() -> frame_support::weights::Weight {
+    if PalletVersion::<T>::get() == types::StorageVersion::V9Struct {
+        migrate_entities::<T>() + migrate_nodes::<T>() + update_pallet_storage_version::<T>()
+    } else {
+        info!(" >>> Unused migration");
+        0
+    }
+}
+
+fn migrate_entities<T: Config>() -> frame_support::weights::Weight {
+    info!(" >>> Migrating entities storage...");
+
+    let mut migrated_count = 0;
+
+    // We transform the storage values from the old into the new format.
+    Entities::<T>::translate::<deprecated::Entity<AccountIdOf<T>>, _>(|k, entity| {
+        info!("     Migrated entity for {:?}...", k);
+
+        let country_name_input: CountryNameInput = BoundedVec::try_from(entity.country).unwrap();
+        let country = match <T as Config>::CountryName::try_from(country_name_input) {
+            Ok(country_name) => country_name,
+            Err(e) => {
+                info!(
+                    "failed to parse country name for entity: {:?}, error: {:?}",
+                    k, e
+                );
+                let default_country_name_input: CountryNameInput =
+                    BoundedVec::try_from(node::DEFAULT_COUNTRY_NAME.to_vec()).unwrap();
+                info!("set default country name for entity");
+                <T as Config>::CountryName::try_from(default_country_name_input).unwrap()
+            }
+        };
+
+        let city_name_input: CityNameInput = BoundedVec::try_from(entity.city).unwrap();
+        let city = match <T as Config>::CityName::try_from(city_name_input) {
+            Ok(city_name) => city_name,
+            Err(e) => {
+                info!(
+                    "failed to parse city name for entity: {:?}, error: {:?}",
+                    k, e
+                );
+                let default_city_name_input: CityNameInput =
+                    BoundedVec::try_from(node::DEFAULT_CITY_NAME.to_vec()).unwrap();
+
+                info!("set default city name for entity");
+                <T as Config>::CityName::try_from(default_city_name_input).unwrap()
+            }
+        };
+
+        let new_entity = TfgridEntity::<T> {
+            version: 2, // ??
```

Comment:
which version here?
@DylanVerstraete

Reply:
Leave it like that, we will be deprecating this versioningen